compiler/src/dotty/tools/dotc/Compiler.scala

```diff
@@ -35,7 +35,8 @@ class Compiler {
   protected def frontendPhases: List[List[Phase]] =
     List(new Parser) ::             // Compiler frontend: scanner, parser
     List(new TyperPhase) ::         // Compiler frontend: namer, typer
-    List(new CheckUnused.PostTyper) ::  // Check for unused elements
+    List(new CheckUnused.PostTyper) :: // Check for unused elements
+    List(new CheckShadowing) :: // Check shadowing elements
```

Comment:
Surely we can merge this miniphase with other mini phases?

Reply:
Yes, that's the plan for linting. But a weird bug appeared when we did that, and we could not diagnose and fix it in a short time frame. I've planned to investigate that, but this version was tested, and we decided to go iteratively while not blocking this feature from releasing.

Reply:
Especially given that a megaphase seems to be a no-op when it is not enabled (isRunnable == false) - it seems to not introduce much cost.